processing/src/main/java/org/apache/druid/query/aggregation/first/MultiStringFirstDimensionVectorAggregator.java

```diff
@@ -0,0 +1,124 @@
+  private long firstTime;
+  private final int maxStringBytes;
+  private final boolean useDefault = NullHandling.replaceWithDefault();
+
+  public MultiStringFirstDimensionVectorAggregator(
+      BaseLongVectorValueSelector timeSelector,
+      MultiValueDimensionVectorSelector valueDimensionVectorSelector,
+      int maxStringBytes
+  )
+  {
+    this.timeSelector = timeSelector;
+    this.valueDimensionVectorSelector = valueDimensionVectorSelector;
+    this.maxStringBytes = maxStringBytes;
+    firstTime = Long.MAX_VALUE;
+  }
+
+  @Override
+  public void init(ByteBuffer buf, int position)
+  {
+    buf.putLong(position, Long.MAX_VALUE);
+    buf.put(
+        position + NumericFirstVectorAggregator.NULL_OFFSET,
+        useDefault ? NullHandling.IS_NOT_NULL_BYTE : NullHandling.IS_NULL_BYTE
+    );
+    buf.putLong(position + NumericFirstVectorAggregator.VALUE_OFFSET, 0);
+  }
+
+  @Override
+  public void aggregate(ByteBuffer buf, int position, int startRow, int endRow)
+  {
+    final long[] timeVector = timeSelector.getLongVector();
+    final IndexedInts[] valueVector = valueDimensionVectorSelector.getRowVector();
+    firstTime = buf.getLong(position);
+    int index = startRow;
+    for (int i = startRow; i < endRow; i++) {
+      if (valueVector[i].get(0) != 0) {
+        index = i;
+        break;
+      }
+    }
+
+    final long earliestTime = timeVector[index];
+    if (earliestTime < firstTime) {
+      firstTime = earliestTime;
+      buf.putLong(position, firstTime);
+      buf.put(position + NumericFirstVectorAggregator.NULL_OFFSET, NullHandling.IS_NOT_NULL_BYTE);
+      buf.putInt(position + NumericFirstVectorAggregator.VALUE_OFFSET, valueVector[index].get(0));
+    }
+  }
+
+  @Override
+  public void aggregate(ByteBuffer buf, int numRows, int[] positions, @Nullable int[] rows, int positionOffset)
+  {
+    long[] timeVector = timeSelector.getLongVector();
+    IndexedInts[] values = valueDimensionVectorSelector.getRowVector();
+    for (int i = 0; i < numRows; i++) {
+      int position = positions[i] + positionOffset;
+      int row = rows == null ? i : rows[i];
+      long firstTime = buf.getLong(position);
+      if (timeVector[row] < firstTime) {
+        firstTime = timeVector[row];
+        buf.putLong(position, firstTime);
+        buf.put(position + NumericFirstVectorAggregator.NULL_OFFSET, NullHandling.IS_NOT_NULL_BYTE);
+        buf.putInt(
+            position + NumericFirstVectorAggregator.VALUE_OFFSET,
+            values[row].size() > 0 ? values[row].get(0) : 0
+        );
+      }
+    }
+  }
+
+  @Nullable
+  @Override
+  public Object get(ByteBuffer buf, int position)
+  {
+    int index = buf.getInt(position + NumericFirstVectorAggregator.VALUE_OFFSET);
+    long earliest = buf.getLong(position);
+    String strValue = valueDimensionVectorSelector.lookupName(index);
+    return new SerializablePairLongString(earliest, StringUtils.chop(strValue, maxStringBytes));
```

Comment:
can this be wrong in the case where nothing was aggregated and id 0 in the dictionary is not null? it seems like we need to check the null byte here and return null if the null byte is set to null (since otherwise it appears as if it will be set to not null)